ccstruct/matrix.h

```diff
@@ -475,26 +463,17 @@ class GENERIC_2D_ARRAY {
     return true;
   }
   // Factored helper to deserialize the size.
-  // If swap is true, assumes a big/little-endian swap is needed.
-  bool DeSerializeSize(bool swap, FILE* fp) {
+  bool DeSerializeSize(FILE* fp) {
     inT32 size1, size2;
     if (fread(&size1, sizeof(size1), 1, fp) != 1) return false;
     if (fread(&size2, sizeof(size2), 1, fp) != 1) return false;
-    if (swap) {
-      ReverseN(&size1, sizeof(size1));
-      ReverseN(&size2, sizeof(size2));
-    }
     Resize(size1, size2, empty_);
```

Comment:
There is a malloc failure and a crash right here if you don't reverse the size to be allocated!